### ISSUE TITLE
3.3.x regression

### ISSUE THREAD

**Issue by panthershark** [neo4j/neo4j#11309]
I'm not sure if this is a driver or server issue, but the more time I spend with the problem the more it seems there could be a server problem so submitting here also.

Related JS driver issue: 
https://github.com/neo4j/neo4j-javascript-driver/issues/339

- Neo4j version: 3.3.0, 3.3.1, 3.3.2, 3.3.3
- Operating system: Mac OS, Alpine Linux in Docker
- API/Driver: JS driver

### Expected behavior
A query should be able to COLLECT and EXTRACT to return a  list of results without error.

### Actual behavior
The query is not applied and it causes latency / db to become un-responsive. (guessing there is a hanging transaction)

### Steps to Reproduce
1. Start a clean neo4j 3.3.3 instance (I tested with 3.3.0 - 3.3.3 with same result)
2. Put the code below into a file - `index.js` would be nice
3. install neo4j driver from npm and run this code

```
const neo4j = require('neo4j-driver').v1;
const auth = neo4j.auth.basic('neo4j', 'dogs');
const db = neo4j.driver('bolt://localhost', auth);

const d1 = {
  name: 'spike',
  breed: 'bulldog'
};

const d2 = {
  name: 'spot',
  breed: 'chihuahua'
};

const session = db.session();

session
  .run(`MATCH (n) DETACH DELETE n`, {
    d1,
    d2
  })
  .then(() =>
    session.run(`CREATE (d1:dog {d1}), (d2:dog {d2})`, {
      d1,
      d2
    })
  )
  .then(() =>
    session.run(
      `MATCH (d:dog)  
     WITH COLLECT(d) as dogs
     FOREACH(d in dogs |
      SET d.name = d.name
    )
     RETURN EXTRACT(d in dogs | d{ .name, .breed }) as dogs
    `,
      {}
    )
  )
  .then((res) => {
    session.close();
    console.log(res);
    process.exit(0);
  })
  .catch((e) => {
    session.close();
    console.log(e);
    process.exit(1);
  });
```

**Comment by Lojjs** [comment]
@tommymessbauer Thanks for reporting! We in the Neo4j Cypher team will investigate this to see if it is a server problem. If it turns out to be a js driver issue, we will hand it over to our driver team.

**Comment by lutovich** [comment]
Hi @tommymessbauer,

Thanks for reporting this issue and providing a reliable way to reproduce it!
It turned out to be a Cypher problem, which should now be fixed by https://github.com/neo4j/neo4j/pull/11347. Fix will be available in the next neo4j 3.3.5 release.

I'm closing this issue but please feel free to update it if necessary.

**Comment by panthershark** [comment]
no problem. Thank you for the quick resolution. We'll upgrade to 3.3.x when this goes live. :)
